### ISSUE TITLE
Tailoring spell doesn't work inside flyout

### ISSUE THREAD

**Issue by Konsafloof** [ScottIngram/UFO#29]
Tailoring spell doesn't work inside of a flyout button but others do
Here's an example without any addons enabled except for UFO
![2024-08-1015-55-06-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/2fad6661-ba05-4997-8ebb-e0bbb53c2b0c)

**Comment by ScottIngram** [comment]
thanks for the troubleshooting and screen-cap.

Ugh, I can't reproduce this.  Initially, I thought this might be related to [use localized strings to fix the 4 broken trade skills](https://github.com/ScottIngram/UFO/commit/0144b8f230aa760d86c08e71f4d588ab246ce1a1) ... but, maybe not?

I don't supposed you can dig up your file from

WOW INSTALLATION -> World of Warcraft/_retail_/WTF/Account/YOUR_ACCOUNT_NAME/SavedVariables/Ufo.lua

so I can see what ID is being assigned to the tailoring prof when you do it.

**Comment by Konsafloof** [comment]
So uh it works now? somehow?,Ill give you today's and the file of 5 days ago, Github won't let me upload it directly
https://www.dropbox.com/scl/fi/n3vxnst6lfcvcjmk5l504/UFO.lua?rlkey=e1ye38bdas82tq21maoqlkrs3&dl=0 <Todays
https://www.dropbox.com/scl/fi/aajij4so7j0lgry8a6k08/UFO-of-81024.lua?rlkey=1dafpvbt2nfbq61pshdmbtzyr&dl=0 < 8/10/24 4pm est

**Comment by Konsafloof** [comment]
Didn't mean to close it sorry

**Comment by ScottIngram** [comment]
Thanks for the files. 
1) They both have the same ID for tailoring: 3908
2) but is different from the one I have: 195126.  I dragged my entry at the beginning of DF.  I did it again just now and got the same ID.

I'm guessing that professions get different IDs in each expansion?

So I changed my ID to match yours, but, then UFO determined that my tailor didn't know it so it filtered it out entirely.

KK.  I'm not sure how to proceed.  Hopefully it keeps working for you.

**Comment by Konsafloof** [comment]
Knowing WoW it was probably just an issue with 11.0.0

**Comment by Konsafloof** [comment]
Okay it started again I just learned Alchemy on one of my alts and it won't open, Looking in the lua file its the correct spellid
[UFO.txt](https://github.com/user-attachments/files/16652373/UFO.txt)

**Comment by Konsafloof** [comment]
It fixes its self over time it seems its working now

**Comment by ScottIngram** [comment]
Just weird. Well,  better than it NOT fixing itself, I guess — “gift horse”

**Comment by Konsafloof** [comment]
Alchemy is doing it again aaagh, Spell id is correct in the Lua file but it hasn't worked since Ive learned TWW Alchemy a few days ago.
[UFO.txt](https://github.com/user-attachments/files/16762029/UFO.txt)

**Comment by ScottIngram** [comment]
kk, ty.  I'll take my alchemist to the trainer and have him learn TWW Alchemy and see what happens.

**Comment by ScottIngram** [comment]
No clues yet.  My alchy had no problems :-( his alchiy spell ID in the config is 195095.

**Comment by Konsafloof** [comment]
> No clues yet. My alchy had no problems :-( his alchiy spell ID in the config is 195095.

Mine is 2259

**Comment by Konsafloof** [comment]
195095 doesn't work but changing it back to 2259 makes it now work?
Have you looked into opening professions by TradeSkillLineID instead of SpellID?
From a few buttons I've made they seem to work more consistent than SpellIDs when messing with professions

**Comment by ScottIngram** [comment]
I get very frustrated when the Bliz API gives me bad data that other parts of the API don't recognize.  In this case, I ask GetCursorInfo() "what is the user dragging around" and it tells me an ID that its other API (the secure button template) may or may not recognize.

So, yeah, I already have some code that is doing that for a handful of the professions that always give bogus IDs.  Maybe I'll assume that the API is always going to give me bogus IDs for all professions rather than cross my fingers and hope I can sometimes trust it.

**Comment by Konsafloof** [comment]
If its possible you could ask "GetProfessions()" for the players current professions then ask "GetProfessionInfo()" for its skilllineID which should always return a valid ID

**Comment by ScottIngram** [comment]
I ran into this briefly one day this week.  After a morning of successfully doing Khaz Algar tailoring and Khaz Algar enchanting,  I logged in later and wah-wah: both were broken.  I reloaded.  Broken.  I exited and relaunched.  Broken.  Later in the day?  Inexplicably fixed again.  I have no Idea what caused it.  I don't know if it is in any way related to the problem that affects Inscription, Jewelcrafting, and Engineering (the button for each of those 3 share the exact same name as their profession whereas all the others ones have names like "Herbalism Journal") 

Yeah.  This is what life as a Bliz API victim is like.  We should form a support group.

Long ago, I wrote code for the 3 problem professions that hardcodes their IDs into a map.  I scrapped it and wrote new, more generic code that will do the same thing but for all professions if any of them start dropping the word "Journal" from the names of their buttons.  Which might(?!) have been happening here.

Thanks, @Konsafloof, for the suggestion of using GetProfessions().  I used it and GetProfessionInfo() to create a mapping from the localized profession name to its TradeSkillLineID.  

This code will break if the Bliz API GetCursorInfo decides to return a different localized profession name than GetProfessionInfo(), which, who fucking knows.

I'll check in the code later and include it in the next release.
